pkg/operator/cassandra/controller/util/util.go

```diff
@@ -0,0 +1,311 @@
+package util
+
+import (
+	"fmt"
+	"github.com/coreos/pkg/capnslog"
+	cassandrarookio "github.com/rook/rook/pkg/apis/cassandra.rook.io"
+	cassandrav1alpha1 "github.com/rook/rook/pkg/apis/cassandra.rook.io/v1alpha1"
+	"github.com/rook/rook/pkg/operator/cassandra/constants"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/selection"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	corelisters "k8s.io/client-go/listers/core/v1"
+)
+
+var logger = capnslog.NewPackageLogger("github.com/rook/rook", "util")
+
+// GetPodsForCluster returns the existing Pods for
+// the given cluster
+func GetPodsForCluster(cluster *cassandrav1alpha1.Cluster, podLister corelisters.PodLister) ([]*corev1.Pod, error) {
+
+	clusterRequirement, err := labels.NewRequirement(constants.ClusterNameLabel, selection.Equals, []string{cluster.Name})
+	if err != nil {
+		logger.Errorf("Error trying to create clusterRequirement")
+		return nil, err
+	}
+	clusterSelector := labels.NewSelector().Add(*clusterRequirement)
+	return podLister.List(clusterSelector)
+
+}
+
+// VerifyOwner checks if the owner Object is the controller
+// of the obj Object and returns an error if it isn't.
+func VerifyOwner(obj, owner metav1.Object) error {
```

Comment:
any functions in this util package that are generally applicable beyond Cassandra and could be moved to the general k8sutil package? (e.g. i don't see anything Cassandra specific in `VerifyOwner`, could be others too)